File: euler023/solution.cs
//Non-abundant sums


 //1. get the abandant numbers
 
    //int[] a=new int[];
    List<int> a = new List<int>();//a dynamic arry is needed!
    //a.Capacity = 28123; //not necessary
    for (int i = 1; i <= 28123; i++)
    {
        int sum = 0;
        for (int j = 2; j <= i / 2; j++)
        {
            if (i % j == 0)
                sum += j;
        }
        if (sum > i)
        {
            a.Add(i);
            //Console.Write(i+" ");
        }
    }

    //2. mark those who can be expressed of 2 abandant numbers.
    Console.WriteLine("Now step 2. ");
    bool[] numB = new bool[28123];//default: false. Need to remove: true.
    //var numVar = Enumerable.Range(1, 28123).ToArray();
    for (int ii = 0; ii <= a.Count - 1; ii++)
    {
        for (int jj = 0; jj <= a.Count - 1; jj++)
        {
            if (a[ii] + a[jj] <= 28123)
            {
                if (numB[a[ii] + a[jj] - 1] == false)
                {
                    numB[a[ii] + a[jj] - 1] = true;
                    //Console.WriteLine("Non-abundant sum: {0}", a[ii] + a[jj]);
                }
            }
        }
    }

    Console.WriteLine("Now step 3. ");
    int sumAllNonAbandants = 0;
    for (int i = 0; i <= 28123-1; i++)
    {
        if (numB[i] == false)
        {
            sumAllNonAbandants += i+1;
        }
    }

    Console.WriteLine("The sum is {0}", sumAllNonAbandants);
}//end of Main()

File: euler023/solution.py
#!/usr/bin/python
#Non-abundant sums

def main():
	#find the abundant numbers in [1,28123)
	abun=[]  #use [] or list() to create a list/array
	for i in range(1, 28123):
		sum=0
		for j in range(2,i/2+1):
			if i%j==0:
				sum=sum+j
		if sum>i:
			abun.append(i)
	# print abun

	#mark numbers that can be expressed by the sum
	mark=[False]*28123
	#i<=j
	l=len(abun)
	for i in range(0,l):
		for j in range(i,l):
			if abun[i]+abun[j]<=28123:
				mark[abun[i]+abun[j]-1]=True


	result=0
	for i in range(0,28123):
		if mark[i]==False:
			result=result+i+1
	print result

main()
